cloudnet-driver/src/main/java/de/dytanic/cloudnet/driver/module/IModuleWrapper.java

```diff
@@ -18,47 +18,160 @@
 
 import de.dytanic.cloudnet.common.document.gson.JsonDocument;
 import java.io.File;
+import java.net.URI;
 import java.net.URL;
 import java.nio.file.Path;
-import java.util.EnumMap;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Unmodifiable;
 
+/**
+ * Represents a wrapper for a module.
+ */
 public interface IModuleWrapper {
 
-  EnumMap<ModuleLifeCycle, List<IModuleTaskEntry>> getModuleTasks();
-
-  IModule getModule();
-
-  ModuleLifeCycle getModuleLifeCycle();
-
-  IModuleProvider getModuleProvider();
-
-  ModuleConfiguration getModuleConfiguration();
-
-  JsonDocument getModuleConfigurationSource();
-
-  ClassLoader getClassLoader();
-
-  IModuleWrapper loadModule();
-
-  IModuleWrapper startModule();
-
-  IModuleWrapper stopModule();
+  /**
+   * Get all module tasks which were detected in the main class of the module.
+   *
+   * @return all module tasks which were detected in the main class of the module.
```

Comment:
Should we perhaps inform that this map is not modifiable, apart from the annotation?